pkg/specgen/volumes.go

```diff
@@ -139,7 +139,13 @@ func GenVolumeMounts(volumeFlag []string) (map[string]spec.Mount, map[string]*Na
 				// This is a overlay volume
 				newOverlayVol := new(OverlayVolume)
 				newOverlayVol.Destination = dest
-				newOverlayVol.Source = src
+				// convert src to absolute path so we don't end up passing
+				// relative values as lowerdir for overlay mounts
+				source, err := filepath.Abs(src)
+				if err != nil {
+					return nil, nil, nil, errors.Wrapf(err, "failed while resolving absolute path for src of overlay mount")
```

Comment:
let's include `src` in the error message, so if there are multiple overlay mounts it is easier to see which one is failing

Reply:
Will that cause a stutter

Reply:
I think it will fail in order like error for the first bad source so stutter should not be there. But can amend if there is any room for stutter.

Reply:
I could not get this filepath.Abs to fail in any way using weird paths, so I am not concerned about this.  Not sure what would cause this to fail, perhaps out of memory or something.

Reply:
it can fail if the current directory is not readable to the user:

```
$ cat /tmp/r.go
package main

import (
	"fmt"
	"path/filepath"
)
func main(){
	_, err := filepath.Abs("foo")
	if err != nil {
		fmt.Printf("ERROR: %v\n", err)
	}
}

$ go build /tmp/r.go; mkdir /tmp/foo; cd /tmp/foo; chmod 0 /tmp/foo; /tmp/r
ERROR: stat .: permission denied
```